lib/check-sanity.js

```diff
@@ -0,0 +1,21 @@
+export default function (point) {
+  checkCoord(point.x);
+  checkCoord(point.y);
+}
+function checkCoord(num) {
+  if (typeof Number.isFininte === 'function') {
+    if (Number.isFininte(num)) {
+      return;
+    }
+    throw new TypeError('coordinates must be finite numbers');
+  }
+  if (typeof num !== 'number') {
+    throw new TypeError('coordinates must be finite numbers');
+  }
+  if (num !== num) {
```

Comment:
this is a check for NaN, basicaly we try to use the new Number.isFinite method which checks that the number isn't nan, that it's not Infinity, and that it's actually a number, if that isn't available we do them seperatly